### PR TITLE
Cleanup SQL in xml/templates/civicrm_state_province.tpl

### PR DESCRIPTION
**Pull request by mlutfy**
Overview
----------------------------------------

Followup to #28025, this is a cleanup of the list of state/provinces XML:

- Removes the hardcoded IDs in the first part, but keeps the minimum at 1000 (even though developers should not rely on IDs, but I'm sure some do)
- Regroups state/provinces from a same country/region together.

Before
----------------------------------------

Many "add missing provinces for XX"

After
----------------------------------------

Mostly grouped by country.

Comments
----------------------------------------

I got bored at some point.

### CODE REVIEW

**Comment by colemanw** [comment]
Oof, this is a hard diff to read but I get the idea.
@mlutfy how about a massive search/replace to delete all the NULL ids from the insert queries? They don't make a lot of sense.

**Comment by mlutfy** [comment]
I guess I could start by doing the NULL cleanup, then cleanup the "add missing provinces" bits in a separate PR.

Any thoughts on the best way to start at 1000? or should I drop that? I guess I could do `alter table .. AUTO_INCREMENT=1000`, then the big insert?

**Comment by colemanw** [comment]
@mlutfy I have no idea why 1000 is the magic number, but sure, that would work.

**Comment by colemanw** [comment]
@mlutfy I'm not saying you have to undo all your work. I know it was a lot. I just think we could *also* delete the NULL ids while we're messing with it.

**Comment by mlutfy** [comment]
I wasn't comfortable with the huge diff, so I had another go at it. I separated into 2 commits, one to remove the IDs, and the other to regroup the "missing" state/provinces.

It's still no fun to review, but at least it's possible to do a few random spot-checks.

(the first commit does have an error, it has a few entries that still had the NULL value, but it was cleaned up in the second commit)

**Comment by mlutfy** [comment]
jenkins, test this please

**Comment by mlutfy** [comment]
This test fail sounds related:

```

not ok 404 - Failure: api\v4\Entity\ContactJoinTest::testCreateWithPrimaryAndBilling_Nigeria
  ---
  message: Failed asserting that 3644 matches expected 3885.
  severity: fail
  data:
    got: 3644
    expected: 3885
  ...
```

**Comment by mlutfy** [comment]
Test fails:

```
not ok 208 - Failure: api_v3_AddressTest::testCreateAddressStateProvinceIDCorrectForCountry
  message: Failed asserting that two strings are equal.
  severity: fail
  data:
    got: "3159"
    expected: "3497"
```

and also:

```
not ok 662 - Failure: CRM_Contact_Import_Parser_ContactTest::testImportCountryStateCounty
  message: should have been valid
  severity: fail
```

**Comment by mlutfy** [comment]
I updated the tests

**Comment by mlutfy** [comment]
Looks like there are more fails:

```
not ok 671 - Error: CRM_Contact_Import_Parser_ContactTest::testImportLocations                                                                          
not ok 844 - Failure: CRM_Contribute_Form_AdditionalPaymentTest::testAddPaymentUsingCreditCardForPartiallyPaidContribution                              
    ---                                                                        
    message: 'Vancouver, AE 1321312 .  not found in  MIME-Version: 1.0'        
    severity: fail                                                             
    ...                     
         
not ok 848 - Failure: CRM_Contribute_Form_AdditionalPaymentTest::testAddPaymentUsingCreditCardForPendingPayLaterContribution                            
    ---                                                                        
    message: 'Vancouver, AE 1321312 .  not found in  MIME-Version: 1.0'        
    severity: fail                                                      
```

File: tests/phpunit/CRM/Contribute/Form/AdditionalPaymentTest.php [hunk 1]
```diff
@@ -372,7 +377,7 @@ public function submitPayment(float $amount, string $mode = NULL, bool $isEmailR
         'cvv2' => 234,
         'credit_card_type' => 'Visa',
         'billing_city-5' => 'Vancouver',
-        'billing_state_province_id-5' => 1059,
+        'billing_state_province_id-5' => $stateProvinceBC['id'],
```
Comment (mlutfy) [inline tests/phpunit/CRM/Contribute/Form/AdditionalPaymentTest.php:380]: In the original xml, 1059 was "Armed Forces" (AE)

**Comment by mlutfy** [comment]
Tests are passing :tada:

**Comment by colemanw** [comment]
This is great @mlutfy. I hope there isn't further fallout from this, but if there is, it's because relying on auto-increment IDs to be the same across all databases is insane! If issues pop up from this in the Civi-verse IMO it's better to fix those problems at the source than continue to support this madness here.

**Comment by mlutfy** [comment]
It only affects new installs too, so in theory it would only affect extensions that hardcoded IDs in a weird way. Thanks for the merge!

**Pull request merged**